### ISSUE TITLE
frontend > 2.100 support

### ISSUE THREAD

**Issue by belka-ew** [jacob-carlborg/dlp#16]
The DMD frontend version used by DLP is starting to show its age. I'm currently have the problem that I can't use std.sumtype. Are there any plans to update the frontend?

I've looked into ddc and there seem to be quite a few code changes and rebasing is not that easy (at least without knowing what has changed).  Is it possible to make DLP work with the official frontend (available as Dub package since 2.100)?

**Comment by jacob-carlborg** [comment]
> The DMD frontend version used by DLP is starting to show its age. I'm currently have the problem that I can't use std.sumtype. Are there any plans to update the frontend?

No, unfortunately there are currently no plans to update the frontend. This project turned out to be an experiment that failed.

> I've looked into ddc and there seem to be quite a few code changes and rebasing is not that easy (at least without knowing what has changed)

There are very few changes in DDC. I think it should be easy.

> Is it possible to make DLP work with the official frontend (available as Dub package since 2.100)?

No, I don't think that would be possible. BTW, I created the Dub package several years ago.

**Comment by linkrope** [comment]
We use infer-attributes regularly at work.
In what way do you think the experiment failed?
Because we are thinking about creating a pull request or in the worst case a fork.

**Comment by jacob-carlborg** [comment]
> In what way do you think the experiment failed?

The experiment was to see if it's possible to build tools using the DMD frontend. Since I had to create a fork it shows that the upstream is not good enough to build tools on. There are other features I would have liked to implement but there's too much trouble with the DMD frontend so I gave up.

> Because we are thinking about creating a pull request or in the worst case a fork.

Sure, I can merge a pull request.
